pkgs/top-level/haskell-packages.nix

```diff
@@ -63,41 +66,41 @@ in rec {
       ghc = compiler.ghc742Binary;
     };
     ghc7103 = callPackage ../development/compilers/ghc/7.10.3.nix rec {
-      bootPkgs = packages.ghc784;
+      bootPkgs = packages.ghc7103Binary;
       inherit (bootPkgs) hscolour;
     };
     ghc802 = callPackage ../development/compilers/ghc/8.0.2.nix rec {
-      bootPkgs = packages.ghc7103;
+      bootPkgs = packages.ghc7103Binary;
       inherit (bootPkgs) hscolour;
       sphinx = pkgs.python27Packages.sphinx;
     };
     ghc822 = callPackage ../development/compilers/ghc/8.2.2.nix rec {
-      bootPkgs = packages.ghc802;
+      bootPkgs = packages.ghc7103Binary;
       inherit (bootPkgs) hscolour alex happy;
       inherit buildPlatform targetPlatform;
       sphinx = pkgs.python3Packages.sphinx;
       selfPkgs = packages.ghc822;
     };
     ghcHEAD = callPackage ../development/compilers/ghc/head.nix rec {
-      bootPkgs = packages.ghc822;
+      bootPkgs = packages.ghc821Binary;
       inherit (bootPkgs) alex happy;
       inherit buildPlatform targetPlatform;
       selfPkgs = packages.ghcHEAD;
     };
     ghcjs = packages.ghc7103.callPackage ../development/compilers/ghcjs {
-      bootPkgs = packages.ghc7103;
+      bootPkgs = packages.ghc821Binary;
```

Comment:
IIRC GHCJS needs older but I forget

Reply:
I haven't bothered since it's broken before this PR.